src/main/java/world/bentobox/bentobox/managers/island/NewIsland.java

```diff
@@ -165,6 +162,12 @@ public void newIsland() {
 
             // Teleport player after this island is built
             plugin.getIslands().homeTeleport(world, user.getPlayer(), true);
+
+            // Delete old island
+            if (oldIsland != null) {
+                // Delete the old island
+                plugin.getIslands().deleteIsland(oldIsland, true);
+            }
```

Comment:
Another bug: The ordering was wrong before. Island was being deleted before the player's island was pasted so they ended up in the acid sea.